### PR TITLE
refactor : separate the different storage pattern processing logic

### PR DESCRIPTION
**Pull request by ph3636**

### CODE REVIEW

File: server/src/main/java/io/seata/server/lock/LockerManagerFactory.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * The type Lock manager factory.
+ *
+ * @author sharajava
+ */
+public class LockerManagerFactory {
+
+    /**
+     * the lock manager
+     */
+    private static LockManager lockManager = EnhancedServiceLoader.load(LockManager.class,
```
Comment (objcoding) [inline server/src/main/java/io/seata/server/lock/LockerManagerFactory.java]: Uppercase static constants.

Comment (ph3636) [reply server/src/main/java/io/seata/server/lock/LockerManagerFactory.java]: done

File: server/src/main/java/io/seata/server/storage/db/lock/DataBaseLockManager.java [hunk 2]
```diff
@@ -44,11 +61,19 @@ public boolean releaseLock(BranchSession branchSession) throws TransactionExcept
...
     @Override
     public boolean releaseGlobalSessionLock(GlobalSession globalSession) throws TransactionException {
-        ArrayList<BranchSession> branchSessions = globalSession.getBranchSessions();
+        List<BranchSession> branchSessions = globalSession.getBranchSessions();
         if (CollectionUtils.isEmpty(branchSessions)) {
             return true;
         }
-        List<Long> branchIds = branchSessions.stream().map(BranchSession::getBranchId).collect(Collectors.toList());
+        List<Long> branchIds = new ArrayList<>(branchSessions.size());
+        boolean notNeedReleaseLock = true;
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/storage/db/lock/DataBaseLockManager.java]: already ensure  SessionHolder.findGlobalSession(xid, false) are not in the case.

File: server/src/main/java/io/seata/server/lock/AbstractLockManager.java [hunk 3]
```diff
@@ -106,7 +111,7 @@ protected Locker getLocker() {
      * @return the locker
      */
     protected Locker getLocker(BranchSession branchSession) {
```
Comment (slievrly) [inline server/src/main/java/io/seata/server/lock/AbstractLockManager.java]: why not abstract method？

**Comment by slievrly** [comment]
1. LockStore  取消SPI，LockStoreDataBaseDAO 代替
2. TransactionStoreManager 取消SPI DataBaseTransactionStoreManager，FileTransactionStoreManager
3. 去除 DefaultSessionManager 并入 FileSessionManager 通过路径判断是否持久化还是内存初始化不同transactionStoreManager

notice：DataBaseLocker(DataSource logStoreDataSource)  需要结合SPI单例修改

File: server/src/main/java/io/seata/server/storage/db/lock/DataBaseLockManager.java (hunk 2 above)
Comment (zjinlei) [inline server/src/main/java/io/seata/server/storage/db/lock/DataBaseLockManager.java]: unnecessary attribute

Comment (zjinlei) [reply server/src/main/java/io/seata/server/storage/db/lock/DataBaseLockManager.java]: lock_key is always blank id db mode

File: server/src/main/java/io/seata/server/lock/AbstractLockManager.java (hunk 3 above)
Comment (ph3636) [reply server/src/main/java/io/seata/server/lock/AbstractLockManager.java]: done

File: server/src/main/java/io/seata/server/storage/db/lock/DataBaseLockManager.java (hunk 2 above)
Comment (ph3636) [reply server/src/main/java/io/seata/server/storage/db/lock/DataBaseLockManager.java]: done

File: server/src/main/java/io/seata/server/storage/db/session/DataBaseSessionManager.java [hunk 4]
```diff
@@ -79,7 +77,7 @@ public DataBaseSessionManager(String name) {
 
     @Override
     public void init() {
-        transactionStoreManager = EnhancedServiceLoader.load(TransactionStoreManager.class, StoreMode.DB.name());
+        transactionStoreManager = new DataBaseTransactionStoreManager();
```
Comment (zjinlei) [inline server/src/main/java/io/seata/server/storage/db/session/DataBaseSessionManager.java]: Will be initialized 4 times by SessionHolder, it should be static.

Comment (ph3636) [reply server/src/main/java/io/seata/server/storage/db/session/DataBaseSessionManager.java]: done

**Pull request merged**